decentralizediso20022/src/main/java/com/centiglobe/decentralizediso20022/application/internal/IntMessageService.java

```diff
@@ -0,0 +1,92 @@
+package com.centiglobe.decentralizediso20022.application.internal;
+
+import com.prowidesoftware.swift.model.mx.AbstractMX;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import com.prowidesoftware.swift.model.mx.BusinessAppHdrV02;
+
+import static com.centiglobe.decentralizediso20022.util.HTTPSCustomTruststore.configureTruststore;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+
+import javax.net.ssl.HttpsURLConnection;
+
+@Profile("internal")
+@Service
+public class IntMessageService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IntMessageService.class);
+
+    @Value("${server.servlet.context-path}")
+    private String CONTEXT_PATH;
+
+    @Value("${server.ssl.trust-store}")
+    private String TRUST_STORE;
+
+    @Value("${server.ssl.trust-store-password}")
+    private String TRUST_PASS;
+
+    /**
+     * Sends an ISO 20022 message using HTTPS to its dedicated endpoint at the recipent host
+     * 
+     * @param mx The ISO 20022 message to send
+     * @return The HTTP response sent by the recipent host
+     * @throws Exception If sending the message failed
+     */
+    public ResponseEntity send(AbstractMX mx) throws Exception {
+        String host = ((BusinessAppHdrV02)mx.getAppHdr()).getTo().getFIId().getFinInstnId().getNm();
+        byte[] encoded = mx.message().getBytes("utf-8");
+
+        // TODO: Explore possibility of dynamic port?
+        URL url = new URL("https://" + host + ":8443" + endpointOf(mx));

```

Comment:
We probably don't want to hardcode the port and have it set in a .properties file.

Reply:
The problem is that we can't really know which port the other bank is using unless we standardize the port, but that could also be restrictive.

Reply:
This might be something we want to address later.

Reply:
But, shouldn't it be fine not to specify the port?
The important part is that HTTPS is used,

Reply:
I added the port from the properties file